packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -76,11 +76,20 @@ export class Arn {
    * the 'scope' is attached to. If all ARN pieces are supplied, the supplied scope
    * can be 'undefined'.
    */
-  public static format(components: ArnComponents, stack: Stack): string {
-    const partition = components.partition !== undefined ? components.partition : stack.partition;
-    const region = components.region !== undefined ? components.region : stack.region;
-    const account = components.account !== undefined ? components.account : stack.account;
-    const sep = components.sep !== undefined ? components.sep : '/';
+  public static format(components: ArnComponents, stack?: Stack): string {
```

Comment:
Can you please add `@param` docs?

And also add a note that says that stack can be omitted only if partition, region and account are defined.